lute/static/js/dict-tabs.js

```diff
@@ -0,0 +1,445 @@
+
+  return urlObj.hostname;
+}
+
+function getFavicon(domain) {
+  return `http://www.google.com/s2/favicons?domain=${domain}`;
+}
+
+/**
+   * Either open a new window, or show the result in the correct frame.
+   */
+function show_lookup_page(dicturl, text, iframe) {
+  // if iframe is provided use that, else it's an external link
+
+  // const is_bing = (dicturl.indexOf('www.bing.com') != -1);
+  // if (is_bing) {
+  //   let use_text = text;
+  //   const binghash = dicturl.replace('https://www.bing.com/images/search?', '');
+  //   const url = `/bing/search/${langid}/${encodeURIComponent(use_text)}/${encodeURIComponent(binghash)}`;
+  //   document.querySelector(`[name="${iframe}"]`).setAttribute("src", url);
+  //   return;
+  // }
+
+  if (iframe) {
+    // console.log(iframe);
+    loadIFrameDictionary(dicturl, text, iframe);
+  } else {
+    // TODO zzfuture fix: fix_language_dict_asterisk
+    // The URL shouldn not be prepended with trash
+    // (e.g. "*http://" means "open an external window", while
+    // "http://" means "this can be opened in an iframe."
+    // Instead, each dict should have an "is_external" property.
+    loadPopupDictionary(dicturl, text);
+  }
+}
+
+function loadIFrameDictionary(dicturl, text, iframe) {
+  const url = get_lookup_url(dicturl, text);
+  iframe.setAttribute("src", url);
+}
+
+function loadPopupDictionary(dicturl, text) {
+  dicturl = dicturl.slice(1);
+  const url = get_lookup_url(dicturl, text);
+  openPopupWindow(url);
+}
+
+function openPopupWindow(url) {
+  window.open(
+    url,
+    'otherwin',
+    'width=800, height=400, scrollbars=yes, menubar=no, resizable=yes, status=no'
+  );
+};
+
+function get_lookup_url(dicturl, term) {
+  let ret = dicturl;
+
+  // Terms are saved with zero-width space between each token;
+  // remove that for dict searches!
+  const zeroWidthSpace = '\u200b';
+  const sqlZWS = '%E2%80%8B';
+  const cleantext = term.
+        replaceAll(zeroWidthSpace, '').
+        replace(/\s+/g, ' ');
+  const searchterm = encodeURIComponent(cleantext).
+        replaceAll(sqlZWS, '');
+  ret = ret.replace('###', searchterm);
+  return ret;
+}
+
+
+function do_image_lookup(text, iframe) {
+  if (LANG_ID == null || LANG_ID == '' || parseInt(LANG_ID) == 0 || text == null || text == '') {
+    alert('Please select a language and enter the term.');
+    return;
+  }
+
+  let use_text = text;
+
+  // If there is a single parent, use that as the basis of the lookup.
+  const parents = get_parents();
+  if (parents.length == 1)
+    use_text = parents[0];
+
+  const raw_bing_url = 'https://www.bing.com/images/search?q=###&form=HDRSC2&first=1&tsc=ImageHoverTitle';
+  const binghash = raw_bing_url.replace('https://www.bing.com/images/search?', '');
+  const url = `/bing/search/${LANG_ID}/${encodeURIComponent(use_text)}/${encodeURIComponent(binghash)}`;
+  iframe.setAttribute("src", url);
+  return;
+}
+
+/** Parents are in the tagify-managed #parentslist input box. */
```

Comment:
Should this function be in here?  get_parents is called from this file as well as from `lute/templates/term/_form.html`.